### PR TITLE
Add fees to claim trie transactions

### PR DESCRIPTION
**Pull request by kaykurokawa**
7f2f918 / reworked to: 1f137fc 
Add new function CalcMinClaimTrieFee to calculate the minimum fee required for claim trie transactions. Works based on MIN_FEE_PER_NAMECLAIM_CHAR , which defines the fee attached per each character of the name in the name claim. Set to 200000 base units or (0.002 LBC) for max fee 0.51 LBC (0.002*255 max length name claim). 

Added test for it in nameclaim_tests.cpp (that we should expand with other tests for nameclaim.cpp) 

8384c04 / reworked to: 12784d4
increase DEFAULT_TRANSACTION_MAXFEE from 0.1 to 1 since we can now get fees over 0.1 LBC

7f2f918 / reworked to: 5c4d8b9
implement CalcMinClaimTrieFee into AcceptToMemoryPool to allow CalcMinClaimTrieFee to take effect when rejecting transactions from the memory pool. Implement CalcMinClaimTrieFee into src/wallet/wallet.cpp so that generated transactions take into consideration the claim trie fee. 

Not sure this is the best way to do it though, ideally the function signature for AcceptToMemoryPool is not affected, but we still need a way to turn off the cacluation of minimum claim trie fee for the tests in claimtrie_tests.cpp and claimtriebranching_tests.cpp, otherwise they break. Will rethink this commit. 

EDIT2: This has been reworked : CalcMinClaimTrieFee now takes as argument minFeePerNameClaimChar . This variable is set as a global in main.cpp along with other globals. This allows it to be set freely to whatever is needed for testing purposes.

### CODE REVIEW

File: src/nameclaim.cpp [hunk 1]
```diff
@@ -173,3 +175,22 @@ size_t ClaimNameSize(const CScript& scriptIn)
         return vvchParams[0].size();
     }
 }
+
+CAmount CalcMinClaimTrieFee(const CTransaction& tx, const CAmount &minFeePerNameClaimChar)
+{
+    CAmount min_fee = 0;
+    BOOST_FOREACH(const CTxOut& txout, tx.vout)
```
Comment (lbrynaut) [inline src/nameclaim.cpp]: Why not skip the loop when minFeePerNameClaimChar is 0?

File: src/nameclaim.h [hunk 2]
```diff
@@ -1,33 +1,45 @@
...
 #include <vector>
 
+// This is the minimum claim fee per character in the name of an OP_CLAIM_NAME command that must
+// be attached to transactions for it to be accepted into the memory pool.
+// Rationale: current implementation of the claim trie uses more memory for longer name claims
+// due to the fact that each chracater is assigned a trie node regardless of whether it contains
+// any claims or not. In the future, we can switch to a radix tree implementation where
+// empty nodes do not take up any memory and the minimum fee can be priced on a per claim
+// basis.
+#define MIN_FEE_PER_NAMECLAIM_CHAR 200000
```
Comment (lbrynaut) [inline src/nameclaim.h]: The PR looks good, and I haven't had as much time to consider the fee structure, but this strikes me as a bit high for a minimum at this point ... although it'll effectively act as a deterrent (perhaps the intended effect).

File: src/nameclaim.cpp (hunk 1 above)
Comment (kaykurokawa) [reply src/nameclaim.cpp]: good idea

**Comment by tzarebczan** [comment]
@kaykurokawa what's the next step here?

**Comment by kaykurokawa** [comment]
Made small change that lbrynaut suggested, rebased and merging.

**Pull request merged**